### PR TITLE
Checkbox atomic component

### PR DESCRIPTION
**Pull request by tiltom**
Resolves https://sovryn.monday.com/boards/2218344956/pulses/2255710988

### CODE REVIEW

File: src/app/components/Checkbox/index.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import React, { useCallback, useMemo } from 'react';
+import styles from './index.module.scss';
+
+export enum CheckboxAlignment {
+  LEFT = 'left',
+  RIGHT = 'right',
+}
+
+type CheckboxProps = {
+  label: string | React.ReactNode;
```
Comment (ShynRou) [inline src/app/components/Checkbox/index.tsx:11]:
```suggestion
  label: React.ReactNode;
```
string is already included in ReactNode

File: src/app/components/Checkbox/index.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import { Checkbox } from './index';
+
+export default {
+  title: 'Atoms/Checkbox',
+  component: Checkbox,
+};
+
+const Template: Story<ComponentProps<typeof Checkbox>> = args => (
+  <Checkbox {...args} />
+);
```
Comment (ShynRou) [inline src/app/components/Checkbox/index.stories.tsx:13]: Having a second interactive template as well might be good. So a template that handles the checked state.

File: src/app/components/Checkbox/index.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  onChange,
+  disabled = false,
+  inline = false,
+  alignment = CheckboxAlignment.LEFT,
+  className,
+}) => {
+  const onClick = useCallback(
+    e => {
+      e.preventDefault();
+      !disabled && onChange();
```
Comment (ShynRou) [inline src/app/components/Checkbox/index.tsx:32]: Please just use an `if`. Also please pass the new value to onChange.

**Review by ShynRou — CHANGES_REQUESTED** [review]

File: src/app/components/Checkbox/index.tsx (hunk 3 above)
Comment (tiltom) [reply src/app/components/Checkbox/index.tsx:32]: There is no need for passing values, `checked` is controlled by a parent component (but good catch with `if`).

File: src/app/components/Checkbox/index.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        className,
+      )}
+    >
+      <label
+        className={classNames(styles.label, {
+          [styles.labelDisabled]: disabled,
+        })}
+        onClick={onClick}
+      >
+        <input type="checkbox" className="tw-hidden" />
```
[the remark this answers is not in the document]
Comment (tiltom) [reply src/app/components/Checkbox/index.tsx:57]: Yes, you're right, `onChange` needs to be used on the input instead of `onClick` on the label.

**Review by creed-victor — DISMISSED** [review]

**Comment by GiladSh05** [comment]
1 small design feedback I have is that personally I would prefer the "V" inside the checkbox when selected to be black

**Comment by tiltom** [comment]
> 1 small design feedback I have is that personally I would prefer the "V" inside the checkbox when selected to be black

I made it black, pls review it again if it looks good (after new deployment is done).

**Review by ShynRou — APPROVED** [review]

**Review by soulBit — APPROVED** [review]

**Pull request merged**